### PR TITLE
crossdock - move params to crossdock.T

### PR DESCRIPTION
**Pull request by breerly**
@yarpc/golang

### CODE REVIEW

File: crossdock-go/client.go [hunk 1]
```diff
@@ -58,12 +60,12 @@ func (h requestHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
-type httpParams struct {
-	Request *http.Request
-}
-
-func (h httpParams) Param(name string) string {
-	return h.Request.FormValue(name)
+// extractParams returns a map of params from url values
+func extractParams(p url.Values) (params Params) {
+	for k, l := range p {
+		for _, v := range l {
```
Comment (kriskowal) [inline crossdock-go/client.go]: this indent looks suspicious.

File: crossdock/main.go [hunk 2]
```diff
@@ -36,25 +36,25 @@ func main() {
...
+		echo.Thrift(t)
 	case "errors":
-		errors.Run(t, ps)
+		errors.Run(t)
 	case "headers":
-		headers.Run(t, ps)
+		headers.Run(t)
 	case "tchclient":
-		tchclient.Run(t, ps)
+		tchclient.Run(t)
```
Comment (kriskowal) [inline crossdock/main.go]: I did not notice the use of abbreviations in behavior names earlier. Was hoping we would run with lower-case without abbreviation.

**Comment by kriskowal** [comment]
:ok:

File: crossdock-go/t.go [hunk 3]
```diff
@@ -86,3 +97,20 @@ func (*entryT) FailNow() {
...
+
+// Param gets a key out of the params map
+func (t entryT) Param(key string) string {
+	return t.params[key]
+}
+
+// SetParams updates the params on the T
+func (t *entryT) SetParams(params Params) {
+	t.behavior = params[BehaviorParam]
+	delete(params, BehaviorParam)
```
Comment (abhinav) [inline crossdock-go/t.go]: Do you still need this?

File: crossdock/main.go (hunk 2 above)
Comment (breerly) [inline crossdock/main.go]: So the behavior name is `BEHAVIOR_TCHCLIENT=...`, so I made the package name match... 

you were hoping for `BEHAVIOR_TCHANNELCLIENT=...` with `tchannelclient.Run(t)`?

**Comment by abhinav** [comment]
LGTM

File: crossdock/main.go (hunk 2 above)
Comment (abhinav) [inline crossdock/main.go]: I wouldn't mind full names for the behaviors. We don't usually type it by-hand anyway.

**Pull request merged**